tests/unit_tests/plugins/modules/readers/test_readers.py

```diff
@@ -0,0 +1,73 @@
+# # # Distribution Statement A. Approved for public release. Distribution unlimited.
+# # #
+# # # Author:
+# # # Naval Research Laboratory, Marine Meteorology Division
+# # #
+# # # This program is free software: you can redistribute it and/or modify it under
+# # # the terms of the NRLMMD License included with this program. This program is
+# # # distributed WITHOUT ANY WARRANTY; without even the implied warranty of
+# # # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the included license
+# # # for more details. If you did not receive the license, for more information see:
+# # # https://github.com/U-S-NRL-Marine-Meteorology-Division/
+
+"""Unit tests on all the readers."""
+from os.path import exists
+from os import scandir
+from importlib.resources import files
+
+import pytest
+from geoips.commandline.log_setup import setup_logging
+from geoips.interfaces import readers
+from geoips.geoips_utils import get_entry_point_group
+
+
+LOG = setup_logging()
+
+
+class TestReaders:
+    """Unit tests every reader."""
+
+    plugin_packages = get_entry_point_group("geoips.plugin_packages")
+    available_readers = []
+    for pkg in plugin_packages:
+        dir_path = str(files(pkg.value) / "plugins/modules/readers/")
+        if exists(dir_path):
+            for reader_path in scandir(dir_path):
+                if reader_path.is_file() and "__init__" not in reader_path.name:
+                    available_readers.append(reader_path.name.split(".")[0])
```

Comment:
@AJThorpe See Evan's updates where he just used geoips.interfaces.readers.get_plugins() - you shouldn't have to include the explicit path to plugins/modules/readers.

Reply:
I was wondering about this. Is there a reason that we might want this divorced from the registries or anything like that? If not, yes, this should be using `geoips.interfaces.readers.get_plugins()`.

Reply:
I don't think we would want these divorced from the registries / interfaces ? We need to test all of geoips with the unit tests,  including the registries and interfaces

Reply:
Sounds good. Then, yes, we should update this to use the get_plugins function from the readers interface.

Reply:
https://github.com/NRLMMD-GEOIPS/geoips/pull/375/files#r1374969868
Looks good!